src/Dialog.tsx

```diff
@@ -68,13 +80,22 @@ export default class Dialog extends React.Component<IDialogPropTypes, any> {
   private sentinelEnd: HTMLElement;
   private bodyIsOverflowing: boolean;
   private scrollbarWidth: number;
+  private content: HTMLElement;
+  private removeMouseUpListener: Function;
+  private removeMouseMoveListener: Function;
 
   componentWillMount() {
     this.inTransition = false;
     this.titleId = `rcDialogTitle${uuid++}`;
   }
   componentDidMount() {
     this.componentDidUpdate({});
+    if (this.props.draggable) {
+      this.removeMouseUpListener = addEventListener(document, 'mouseup', this.docMouseUp).remove;
+      let rect =  this.content.getBoundingClientRect();
+      this.position.initX = (rect as any).x;
+      this.position.initY = (rect as any).y;
+    }
```

Comment:
We should create a HOC named `<Draggable>` instead add so many code in lifecycle method of Dialog.

Reply:
emmm,it sounds reasonable,I'll give it a try.